apps/condo/domains/common/utils/date.spec.js

```diff
@@ -35,4 +37,116 @@ describe('Date helper tests', () => {
             expect(nextMonthStart).toStrictEqual('2022-02-01')
         })
     })
+
+    describe('clearDateStr', () => {
+
+        it('should return sanitized date string if input contains extraneous characters', () => {
+            const input = '2021-05-04T12:34:56Z!!!'
+            const output = '2021-05-04T12:34:56Z'
+            expect(clearDateStr(input)).toBe(output)
+        })
+
+        it('should return null for empty string', () => {
+            const input = ''
+            expect(clearDateStr(input)).toBeNull()
+        })
+
+        it('should return null for null input', () => {
+            expect(clearDateStr(null)).toBeNull()
+        })
+
+        it('should return null for undefined input', () => {
+            expect(clearDateStr(undefined)).toBeNull()
+        })
+
+        it('should return null for non-string input (number)', () => {
+            const input = 12345
+            expect(clearDateStr(input)).toBeNull()
+        })
+
+        it('should return null for non-string input (object)', () => {
+            const input = { date: '2021-05-04' }
+            expect(clearDateStr(input)).toBeNull()
+        })
+
+        it('should return null for a string with only invalid characters', () => {
+            const input = '!!!@#@#$$%'
+            expect(clearDateStr(input)).toBeNull()
```

Comment:
```suggestion
            expect(clearDateStr(input)).toEqual('')
```

Reply:
I give you a string, you Promise to clean it. Why you return me nothing? I expect cleared string